### PR TITLE
chore: dm support

### PR DESCRIPTION
**Pull request by catngh**
**What does this PR do?**

-  support command in DM for `bals` and `withdraw`
Screen Shot 2022-08-22 at 18 13 54
Screen Shot 2022-08-22 at 18 14 20

### CODE REVIEW

**Comment by trkhoi** [comment]
fix lint

File: src/commands/defi/balances.ts [hunk 1]
```diff
@@ -22,6 +22,9 @@ const command: Command = {
   brief: "Show your balances of the supported tokens",
   category: "Defi",
   run: async function balances(msg: Message) {
+    if (msg.channel.type == "DM") {
```
Comment (anhnh12) [inline src/commands/defi/balances.ts:25]: ===

File: src/commands/defi/balances.ts [hunk 2]
```diff
@@ -83,9 +86,18 @@ const command: Command = {
       `${getEmoji("money")} \`$${roundFloatNumber(totalBalanceInUSD, 4)}\``
     )
 
-    await msg.author.send({
-      embeds: [embed],
-    })
+    if (msg.channel.type == "DM") {
```
Comment (anhnh12) [inline src/commands/defi/balances.ts:89]: ===

File: src/adapters/defi.ts [hunk 3]
```diff
@@ -130,7 +130,10 @@ class Defi {
     })
 
     const json = await resp.json()
-    if (json.error !== undefined) {
+    if (
```
Comment (anhnh12) [inline src/adapters/defi.ts:133]: should create `InsufficientBalanceError` and throw that if error message includes "balance is not enough"
check `DiscordWalletTransferError.ts` for reference

File: src/commands/defi/withdraw.ts [hunk 4]
```diff
@@ -37,6 +37,19 @@ async function getDestinationAddress(
 async function withdraw(msg: Message, args: string[]) {
   const payload = await Defi.getTransferPayload(msg, args)
   const data = await Defi.discordWalletWithdraw(JSON.stringify(payload))
+  if (data.error.includes("balance is not enough")) {
```
Comment (anhnh12) [inline src/commands/defi/withdraw.ts:40]: should create `InsufficientBalanceError` and throw that if error message includes "balance is not enough"
check `DiscordWalletTransferError.ts` for reference

**Review by anhnh12 — CHANGES_REQUESTED** [review]

**Review by trkhoi — APPROVED** [review]

**Review by anhnh12 — APPROVED** [review]

**Pull request merged**